zipkin-server/src/main/java/zipkin2/server/internal/banner/ZipkinBanner.java

```diff
@@ -13,35 +13,44 @@
  */
 package zipkin2.server.internal.banner;
 
-import java.util.List;
-import org.springframework.boot.ResourceBanner;
+import java.io.InputStream;
+import java.io.PrintStream;
+import org.springframework.boot.Banner;
+import org.springframework.boot.ansi.AnsiElement;
+import org.springframework.boot.ansi.AnsiOutput;
+import org.springframework.boot.ansi.AnsiStyle;
 import org.springframework.core.env.Environment;
-import org.springframework.core.env.MutablePropertySources;
-import org.springframework.core.env.PropertyResolver;
-import org.springframework.core.env.PropertySourcesPropertyResolver;
 import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.StreamUtils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 /**
- * <code>Banner</code> implemetation for Ansi256 Color<br>
- * TODO: This class should be deleted when this feature is provided by Spring Boot
- * @see <a href="https://github.com/spring-projects/spring-boot/pull/18264">https://github.com/spring-projects/spring-boot/pull/18264</a>
+ * More efficient Banner implementation which doesn't use property sources as variables are expanded
+ * at compile time using Maven resource filtering.
  */
-public class ZipkinBanner extends ResourceBanner {
+public class ZipkinBanner implements Banner {
+  static final AnsiElement ZIPKIN_ORANGE = new AnsiElement() {
+    @Override public String toString() {
+      return "38;5;208"; // Ansi 256 color code 208 (orange)
+    }
+  };
 
-  public ZipkinBanner() {
-    super(new ClassPathResource("zipkin.txt"));
-  }
+  final Resource resource = new ClassPathResource("zipkin.txt");
```

Comment:
Can the banner be printed twice? I guess it can be a local. Or otherwise static

Reply:
sgtm